scripts/skills/actives/rf_deep_impact_skill.nut

```diff
@@ -1,149 +1,79 @@
 
-		return ::Reforged.Mod.Tooltips.parseString("Costs as if your primary attack had a base [Fatigue|Concept.Fatigue] cost of " + ::MSU.Text.colorNegative(this.m.FatigueCost));
-	}
+		local effect = ::new("scripts/skills/effects/rf_deep_impact_effect");
+		effect.m.Container = ::MSU.getDummyPlayer().getSkills();
 
-	function getTooltip()
-	{
-		local ret = this.skill.getDefaultUtilityTooltip();
-		if (this.getContainer().getActor().getID() == ::MSU.getDummyPlayer().getID())
-		{
-			ret.push({
-				id = 10,
-				type = "text",
-				icon = "ui/icons/special.png",
-				text = ::Reforged.Mod.Tooltips.parseString("Perform your primary attack on the target and if successful, apply the [Deep Impact|Skill+rf_deep_impact_effect] effect")
-			});
-			return ret;
-		}
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("If the attack is successful and does damage to [Hitpoints|Concept.Hitpoints] the target receives the [Deep Impact|Skill+rf_deep_impact_effect] effect"),
+			children = effect.getTooltip().slice(2) // slice 2 to remove name and description
+		});
 
-		local aoo = this.getContainer().getAttackOfOpportunity();
-		if (aoo == null || !this.isSkillValid(aoo))
-		{
-			ret.push({
-				id = 20,
-				type = "text",
-				icon = "ui/icons/warning.png",
-				text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorNegative("Requires a valid attack"))
-			});
-		}
-		else
+		effect.m.Container = null;
+
+		if (this.m.TwoHandedAPCostAdd != 0 && ::MSU.isEqual(this.getContainer().getActor(), ::MSU.getDummyPlayer()))
 		{
 			ret.push({
-				id = 10,
+				id = 11,
 				type = "text",
-				icon = "ui/icons/special.png",
-				text = ::Reforged.Mod.Tooltips.parseString(format("Perform a %s on the target and if successful, apply the [Deep Impact|Skill+rf_deep_impact_effect] effect", ::Reforged.NestedTooltips.getNestedSkillName(aoo)))
+				icon = "ui/icons/action_points.png",
+				text = ::Reforged.Mod.Tooltips.parseString("Costs " + ::MSU.Text.colorizeValue(this.m.TwoHandedAPCostAdd, {AddSign = true, InvertColor = true}) + " [Action Points|Concept.ActionPoints]with two-handed hammers")
 			});
 		}
 
 		return ret;
 	}
 
-	function onAfterUpdate( _properties )
-	{
-		local aoo = this.getContainer().getAttackOfOpportunity();
-		if (aoo != null && this.isSkillValid(aoo))
-		{
-			this.m.ActionPointCost = aoo.m.ActionPointCost;
-			this.m.FatigueCostMult = aoo.m.FatigueCostMult;
-			this.m.MinRange = aoo.m.MinRange;
-			this.m.MaxRange = aoo.m.MaxRange;
-		}
-	}
-
-	// MSU function
-	// overwrite to return AOO damage type instead of our own
-	function getDamageType()
+	function onAdded()
 	{
-		local aoo = this.getContainer().getAttackOfOpportunity();
-		return aoo != null && this.isSkillValid(aoo) ? aoo.getDamageType() : this.skill.getDamageType();
+		this.setBaseValue("ActionPointCost", this.getBaseValue("ActionPointCost") + this.m.TwoHandedAPCostAdd);
```

Comment:
This looks like you always add the two-handed cost to the skill, even if the character is holding a one-handed weapon